File: pkg/model/warnings.go
package model

import "time"

// QueryWarning is a machine-readable warning attached to a flows query response.
// Frontend maps Code to i18n strings; extra fields support interpolation.
type QueryWarning struct {
	Code           string     `json:"code"`
	Peer           string     `json:"peer,omitempty"`
	BufferOldest   *time.Time `json:"bufferOldest,omitempty"`
	BufferNewest   *time.Time `json:"bufferNewest,omitempty"`
	RequestedStart *time.Time `json:"requestedStart,omitempty"`
	RequestedEnd   *time.Time `json:"requestedEnd,omitempty"`
	Message        string     `json:"message,omitempty"`
}

File: web/src/model/warnings.ts
export type WarningType = 'slow' | 'cantfetchdrops' | 'rawFlowsBufferOnly' | 'peerQueryFailed';

export interface Warning {
  summary: string;
  details?: string;
  type: WarningType;
}

/** Machine-readable warning from backend AggregatedQueryResponse.warnings */
export interface QueryWarning {
  code: string;
  peer?: string;
  bufferOldest?: string;
  bufferNewest?: string;
  requestedStart?: string;
  requestedEnd?: string;
  message?: string;
}

export const RAW_FLOWS_BUFFER_ONLY = 'RAW_FLOWS_BUFFER_ONLY';
export const PEER_QUERY_FAILED = 'PEER_QUERY_FAILED';
